plugin-server/src/worker/ingestion/ingest-event.ts

```diff
@@ -60,3 +62,48 @@ export async function ingestEvent(hub: Hub, event: PluginEvent): Promise<IngestE
         clearTimeout(timeout)
     }
 }
+
+export async function ingestBufferEvent(hub: Hub, event: PreIngestionEvent): Promise<IngestEventResponse> {
+    const person = await hub.db.fetchPerson(event.teamId, event.distinctId)
+    const [, eventId, elements] = await hub.eventsProcessor.createEvent(event)
+    const actionMatches = await handleActionMatches(hub, event as any, '', eventId, elements, person)
+    return { actionMatches, success: true }
+}
+
+async function handleActionMatches(
+    hub: Hub,
+    event: PluginEvent,
+    siteUrl: string,
+    eventId?: number,
+    elements?: Element[],
+    person?: Person
+): Promise<Action[]> {
+    let actionMatches: Action[] = []
+
+    actionMatches = await hub.actionMatcher.match(event, person, elements)
+    await hub.hookCannon.findAndFireHooks(event, person, siteUrl, actionMatches)
+
+    if (actionMatches.length && eventId !== undefined) {
+        await hub.db.registerActionMatch(eventId, actionMatches)
+    }
+
+    return actionMatches
+}
+
+// context: https://github.com/PostHog/posthog/issues/9182
+// TL;DR: events from a recently created non-anonymous person are sent to a buffer
+// because their person_id might change. We merge based on the person_id of the anonymous user
+// so ingestion is delayed for those events to increase our chances of getting person_id correctly
+function shouldSendEventToBuffer(hub: Hub, event: PreIngestionEvent, person?: Person) {
+    const isAnonymousEvent =
+        event.properties && event.properties['$device_id'] && event.distinctId === event.properties['$device_id']
+    const isRecentPerson = !person || DateTime.now().diff(person.created_at).seconds < hub.BUFFER_CONVERSION_SECONDS
```

Comment:
@tiina303

Reply:
btw we also have https://github.com/PostHog/posthog/blob/c776ee3583a62b4ef0babdb7a5e192f3d45c07be/plugin-server/src/worker/ingestion/person-manager.ts#L18